compiler/GHC/Parser/Lexer.x

```diff
@@ -713,6 +720,10 @@ data Token
   | ITanyclass
   | ITvia
 
+  -- QualifiedDo
+  | ITqdo FastString
+  | ITqmdo FastString
```

Comment:
These new constructors (`ITqdo` and `ITqmdo`) need to be added to Haddock in `utils/haddock/haddock-api/src/Haddock/Backends/Hyperlinker/Parser.hs:176:3`. However, that is an upstream library, so I'm not sure how we synchronize that. It only warns of unhandled constructors though.

Reply:
This still holds, except now it causes a compilation error in haddock in stead of a warning. How do we sync the changes?

Reply:
Haddock is complicated. The protocol is to make a branch (I've given you access to Tweag's fork so that you can push it there) and use this local branch for your development. Haddock is in a submodule, you have to edit the `.gitmodules` file to point to Tweag's fork, then `git submodule sync .` then `git submodule update`, basically.

When the MR is accepted on gitlab, there is a special branch to push the haddock patch to in the upstream repository. Then you merge the MR on GHC's master, then the change ends up on Haddock's master somehow.

I may have mixed up some of this.

Reply:
Thanks Arnaud. I saw this comment after merging. Did this instructions with a new fork I made of my own. Could change it to tweag's if it has any relevance.